### PR TITLE
Move Services into ServiceProvider

### PR DESCRIPTION
**Pull request by henrikbjorn**
Some time ago we added KernelServiceProvider and RoutingServiceProvider. I have taken the rest of the services defined directly in the Silex\Application class and moved them into those two service providers.

This makes Application more of a delegation class.

### CODE REVIEW

File: src/Silex/Provider/RoutingServiceProvider.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
  * (c) Fabien Potencier <fabien@symfony.com>
  *
  * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
+ * file that was distributed with app source code.
```
Comment (fabpot) [inline src/Silex/Provider/RoutingServiceProvider.php]: probably wrong

File: src/Silex/Provider/KernelServiceProvider.php [hunk 2]
```diff
@@ -2,26 +2,61 @@
...
 class KernelServiceProvider implements ServiceProviderInterface, EventListenerProviderInterface
 {
     /**
      * {@inheritdoc}
      */
-    public function register(Container $pimple)
+    public function register(Container $app)
     {
-        $pimple['dispatcher'] = function () {
+        $app['controllers'] = function ($app) {
```
Comment (fabpot) [inline src/Silex/Provider/KernelServiceProvider.php]: This one probably belongs to the router

File: src/Silex/Provider/KernelServiceProvider.php [hunk 3]
```diff
@@ -2,26 +2,61 @@
...
+            return $app['controllers_factory'];
+        };
+
+        $app['controllers_factory'] = $app->factory(function ($app) {
```
Comment (fabpot) [inline src/Silex/Provider/KernelServiceProvider.php]: this one as well, no?

Comment (henrikbjorn) [inline src/Silex/Provider/KernelServiceProvider.php]: They have been moved

File: src/Silex/Provider/RoutingServiceProvider.php (hunk 1 above)
Comment (henrikbjorn) [inline src/Silex/Provider/RoutingServiceProvider.php]: changed back.

**Comment by henrikbjorn** [comment]
I moved the two services and updated a whitepsace issue

**Comment by henrikbjorn** [comment]
Rebased and the build is now green.

**Comment by hhamon** [comment]
Nice work :)

**Comment by Hikariii** [comment]
+1

**Comment by stof** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @henrikbjorn.

**Pull request closed without merging**